src/key.rs

```diff
@@ -232,55 +232,75 @@ impl SecretKey {
         }
     }
 
-    #[inline]
     /// Adds one secret key to another, modulo the curve order.
     ///
     /// # Errors
     ///
     /// Returns an error if the resulting key would be invalid or if the tweak was not a 32-byte
     /// length slice.
-    pub fn add_assign(
-        &mut self,
-        other: &[u8],
-    ) -> Result<(), Error> {
-        if other.len() != 32 {
+    #[inline]
+    #[deprecated(since = "TODO: Set this prior to release", note = "Use add_tweak instead")]
+    pub fn add_assign(&mut self, other: &[u8]) -> Result<(), Error> {
+        *self = self.add_tweak(other)?;
+        Ok(())
+    }
+
+    /// Tweaks a [`SecretKey`] by adding `tweak` modulo the curve order.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the resulting key would be invalid or if the tweak was not a 32-byte
```

Comment:
In the long run, it might be better to create a new type: https://docs.rs/secp256k1-zkp/0.6.0/secp256k1_zkp/struct.Tweak.html#

I slightly prefer the name `Scalar` tough.